transcom/mymove
# B 17910 h&al sc creates or modifies a shipment int

## [Agility B-17910](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3a867954)

## Summary

I have verified the items in the H&AL are correctly shown, and have added the downstream action of the closeout office added to the H&AL, as that is part of the flow for editing/creating a move for the SM.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers
(Playwright test for the SM creating a move and the SC getting to the H&AL page not included)

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the
2. Login as a service member
3. Create a ppm
4. Log in as SC
5. Navigate to the move that was newly created
6. Edit details of the move
7. Observe the changes populate properly in the audit history tab
8. Create a new PPM for the service member
9. Observe the changes populate in the audit history tab

### Frontend


### Backend


### Database

The history fetcher sql file has changed
https://github.com/transcom/mymove/blob/2d9d1bbeb70cdb73a8283892c5265953b539e329/pkg/assets/sql_scripts/move_history_fetcher.sql
```javascript
move_logs AS (
  SELECT
    audit_history.*,
    jsonb_agg(jsonb_strip_nulls(
      jsonb_build_object(
        'closeout_office_name',
        (SELECT transportation_offices.name FROM transportation_offices WHERE transportation_offices.id = uuid(c.closeout_office_id))
      ))
    )::TEXT AS context,
    NULL AS context_id
  FROM
    audit_history
  JOIN move ON audit_history.object_id = move.id
  JOIN jsonb_to_record(audit_history.changed_data) as c(closeout_office_id TEXT) on TRUE
  WHERE audit_history.table_name = 'moves' group by audit_history.id
),

```


#### Any new migrations/schema changes:


## Screenshots
